File: python-framework/demo_mcp_code/mcp_server.py

# Demo MCP Server Python Code mit Fehlern
from flask import Flask, jsonify
import json

app = Flask(__name__)

def calculate_average(numbers):
    # Fehler: Division durch Null möglich
    return sum(numbers) / len(numbers)

def process_user_data(user_data):
    # Fehler: Null-Check fehlt
    return user_data.get('name').upper()

@app.route('/api/data')
def get_data():
    try:
        # Fehler: undefined_variable
        return jsonify({'result': undefined_result})
    except Exception as e:
        return jsonify({'error': str(e)})

@app.route('/health')
def health():
    return jsonify({'status': 'healthy'})

if __name__ == '__main__':
    app.run(debug=True, port=5000)

File: python-framework/demo_mcp_code/mcp_server.js

// Demo MCP Server Code mit Fehlern
const express = require('express');
const app = express();

function calculateTotal(items) {
    let total = 0;
    for (let item of items) {
        // Fehler: Null-Check fehlt
        total += item.price;
    }
    return total
} // Fehler: Fehlender Semicolon

function processData(data) {
    // Fehler: Typo - sollte "length" sein
    return data.lenght;
}

// Undefined function call
function main() {
    const items = [null, {price: 10}, {price: 20}];
    const total = calculateSum(items); // Fehler: calculateSum existiert nicht
    console.log('Total:', total);
}

app.get('/health', (req, res) => {
    res.json({status: 'ok'});
});

app.listen(3000, () => {
    console.log('MCP Server running on port 3000');
});
